### PR TITLE
add test for non-root rotations

### PR DESCRIPTION
**Pull request by AdamKorcz**
Migrates the `test_non_root_rotations` test from python-tuf.

### CODE REVIEW

File: tuf_conformance/test_updater_key_rotations.py [hunk 1]
```diff
@@ -145,3 +173,61 @@ def test_root_rotation(client: ClientRunner,
...
+
+        repo.root.version += 1
+        repo.publish_root()
+
+        # run client workflow, assert success/failure
+        expected_error = md_version.res
+        if expected_error:
+            assert client.refresh(init_data) == 0
+
+            # Call fetch_metadata to sign metadata with new keys
```
Comment (jku) [inline tuf_conformance/test_updater_key_rotations.py:224]: nit: this comment is likely outdated, we call fetch_metadata() here to be able to do the comparison below not specifically to sign

File: tuf_conformance/test_updater_key_rotations.py [hunk 2]
```diff
@@ -145,3 +173,61 @@ def test_root_rotation(client: ClientRunner,
...
+
+        for i in md_version.sigs:
+            repo.add_signer(role, signers[i])
+
+        repo.root.version += 1
+        repo.publish_root()
+
+        # run client workflow, assert success/failure
+        expected_error = md_version.res
+        if expected_error:
```
Comment (jku) [inline tuf_conformance/test_updater_key_rotations.py:221]: naming is not great: when the test is expecting success, `expected_error==True`

**Review by jku — APPROVED** [review]
looks pretty good, thanks. There is a variable name that could still be fixed but I'm fine with this as well

File: tuf_conformance/test_updater_key_rotations.py (hunk 1 above)
Comment (AdamKorcz) [reply tuf_conformance/test_updater_key_rotations.py]: https://github.com/theupdateframework/tuf-conformance/pull/104/commits/bdc11da0cac16015bcca384c369e94c58ee36ef4

File: tuf_conformance/test_updater_key_rotations.py (hunk 2 above)
Comment (AdamKorcz) [reply tuf_conformance/test_updater_key_rotations.py]: https://github.com/theupdateframework/tuf-conformance/pull/104/commits/bdc11da0cac16015bcca384c369e94c58ee36ef4

**Pull request merged**